packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -270,10 +270,22 @@ export async function getDynamicEventType(
   });
 }
 
-export function getRegularOrDynamicEventType(
+export async function getRegularOrDynamicEventType(
   input: TGetScheduleInputSchema,
   organizationDetails: { currentOrgDomain: string | null; isValidOrgDomain: boolean }
 ) {
+  if (input.bookerEmail) {
+    const bookerUser = await prisma.user.findUnique({
+      where: { email: input.bookerEmail },
+      select: { username: true },
+    });
+
+    // opt for dynamic booking flow if booker is a cal user
+    if (bookerUser?.username && input.eventTypeSlug) {
+      input.usernameList?.push(bookerUser.username);
+    }
+  }
+
```

Comment:
Hey @SySagar You do the similar changes as your previous pr. Why are you opt dynamic booking flow?

Reply:
actually I wanted to ask this but you closed th pr. I went for this approch because I found out that cal users are only able to do dynamic meeting. Thats why what I found was that there was username list which was only getting one user. SO on updating that list it worked . So went for this approch.

Reply:
I don't think this is the right way to solve. Did you test that locally?

Reply:
The earlier pr worked as expected however some test were failing so I redo this. I just need a bit time  to put the loom video and some basic tests

Reply:
I think Currently cal assumes that the booker is always a non cal user (is booker is an outsider)

This PR now checks for that if the booker is a cal user or not as suggested in the original issue and then opts for dynamic booking flow where their availability slots are calculated later on in the dynamic booking flow logics

Dynamic booking flow requires multiple hosts (or cal users)

Reply:
@SySagar Can you attach a loom video to show if the PR works?